src/System.Xml.ReaderWriter/src/System/Xml/Core/XmlTextReaderImplAsync.cs

```diff
@@ -74,6 +76,43 @@ private Task FinishInitAsync()
             }
         }
 
+        private async Task FinishInitUriStringAsync()
+        {
+            // Substitutes removed XmlResolver.GetEntityAsync
+            Task<object> getEntityAsync = Task.Run(() =>
```

Comment:
I realize you're just restoring existing code, but do you know why that code was using Task.Run here?  It's scheduling a task that it just then awaits.